backend/app/worker/photogrammetry/images_to_point_cloud.py

```diff
@@ -16,6 +19,129 @@ def remove_if_exists(path):
+        ctypes.windll.kernel32.GlobalMemoryStatusEx(ctypes.byref(stat))
+        return int(stat.ullAvailPhys / 1024 / 1024)
+    else:
+        # Linux/macOS implementation
+        with os.popen("free -t -m") as fp:
+            lines = fp.readlines()
+        if not lines:
+            return int(psutil.virtual_memory().available / 1024 / 1024)
+        available_mem = int(lines[1].split()[6])
+        return available_mem
+
+def get_cpu_count() -> int:
+    """Returns the number of available CPU cores"""
+    return psutil.cpu_count(logical=False) or 1  
+
+def get_max_image_resolution(images_dir: str):
+    """Get the maximum resolution of images in the directory"""
+    max_width, max_height = 0, 0
+    try:
+        for filename in os.listdir(images_dir):
+            if filename.lower().endswith(('.jpg', '.jpeg', '.png', '.tif', '.tiff')):
+                img_path = os.path.join(images_dir, filename)
+                img = cv2.imread(img_path)
+                if img is not None:
+                    height, width = img.shape[:2]
+                    max_width = max(max_width, width)
+                    max_height = max(max_height, height)
+    except Exception as e:
+        logger.warning(f"Error reading image dimensions: {e}")
+        return (1920, 1080)
+    
+    if max_width == 0 or max_height == 0:
+        return (1920, 1080)
+    
+    return (max_width, max_height)
+
+def calculate_resource_allocation(images_dir: str):
+    """Calculate optimal resource allocation based on system capabilities and image resolution"""
+    available_memory_mb = memory_available()
+    cpu_count = get_cpu_count()
+
+    max_width, max_height = get_max_image_resolution(images_dir)
+    max_mp = (max_width * max_height) / 1_000_000  # Megapixels
+    
+    logger.info(f"System has {cpu_count} CPU cores and {available_memory_mb}MB available memory")
+    logger.info(f"Maximum image resolution: {max_width}x{max_height} ({max_mp:.2f}MP)")
+    
+
+    #TODO: Adjust these values based on benchmarking  specific workload
+    # memory_per_process_mb = 2000 + (20 * max_mp) 
+    # print(f"Memory per process: {memory_per_process_mb}MB")
+    # memory_limited_processes = max(1, int(available_memory_mb / memory_per_process_mb))
+    # logger.info(f"Memory-limited processes: {memory_limited_processes}")
+    
+    processes = max(1, cpu_count)
+    logger.info(f"Calculated processes: {processes}")
+    depthmap_resolution = 2048  
+    feature_process_size = 2048
+    if processes > 16 and available_memory_mb > 16000:
+        depthmap_resolution = max(4096, max(max_width, max_height))
+        feature_process_size = 4096
+    elif processes > 8 and available_memory_mb > 8000:
+        depthmap_resolution = max(2048, max(max_width, max_height)/2)
+        feature_process_size = 2048
+    else:
+        depthmap_resolution = max(1024, max(max_width, max_height)/4)
+
+    
+    depthmap_processes = max(1, int(processes / 2))  
+
```

Comment:
add a statement to check if the feature_process_size and depthmap_resolution are not greater than max(max_width, max_height)